src/components/tasks/TaskRunsPanel.vue

```diff
@@ -0,0 +1,186 @@
+        </v-virtual-scroll>
+      </div>
+      <v-divider />
+      <v-list-item :title="`Last updated: ${lastUpdatedString}`">
+        <template #append>
+          <v-progress-circular
+            v-if="taskRuns.isLoading.value"
+            size="20"
+            indeterminate
+          />
+          <v-btn
+            v-else
+            density="compact"
+            variant="plain"
+            icon="mdi-refresh"
+            @click="taskRuns.fetch()"
+          />
+        </template>
+      </v-list-item>
+    </div>
+  </Teleport>
+</template>
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue'
+
+import { RelativePeriod } from '@/lib/period'
+import { TaskRun, TaskStatus } from '@/lib/taskruns'
+
+import { useTaskRuns } from '@/services/useTasksRuns'
+
+import { useAvailableWorkflowsStore } from '@/stores/availableWorkflows'
+
+import TaskStatusFilterControl from './TaskStatusFilterControl.vue'
+import TaskRunSummary from './TaskRunSummary.vue'
+import WorkflowFilterControl from './WorkflowFilterControl.vue'
+import PeriodFilterControl from './PeriodFilterControl.vue'
+import { useWhatIfTemplates } from '@/services/useWhatIfTemplate'
+import { configManager } from '@/services/application-config'
+
+const availableWorkflowsStore = useAvailableWorkflowsStore()
+
+const isPanelOpen = ref(false)
+
+const selectedWorkflowIds = ref<string[]>(availableWorkflowsStore.workflowIds)
+const expandedItems = ref<Record<string, boolean>>({})
+
+// Automatically select preferred workflow IDs if the user has not changed the
+// filter manually yet, otherwise, leave the filter as-is.
+let hasChangedWorkflowsFilterManually = false
+const flagManuallyChanged = () => (hasChangedWorkflowsFilterManually = true)
+watch(
+  () => availableWorkflowsStore.preferredWorkflowIds,
+  (preferredWorkflowIds) => {
+    if (hasChangedWorkflowsFilterManually) return
+    if (preferredWorkflowIds.length === 0) {
+      // If we have no preferred workflow IDs, select all.
+      selectedWorkflowIds.value = availableWorkflowsStore.workflowIds
+    } else {
+      // Otherwise, show only the preferred workflow IDs.
+      selectedWorkflowIds.value = preferredWorkflowIds
+    }
+  },
+)
+
+// The request for the available workflows may still be pending while this
+// component is created, so we may initialise the default selected workflows
+// with an empty array. To fix this, we reinitialise our selection upon changes
+// in the workflows, which occur when they have been fetched.
+watch(
+  () => availableWorkflowsStore.workflowIds,
+  (fetchedWorkflowIds) => {
+    selectedWorkflowIds.value = fetchedWorkflowIds
+  },
+  { once: true },
+)
+
+// Select all task statuses by default.
+const selectedTaskStatuses = ref<TaskStatus[]>(Object.values(TaskStatus))
+
+// Look 2 hours back by default..
+const period = ref<RelativePeriod | null>({
+  startOffsetSeconds: -2 * 60 * 60,
+  endOffsetSeconds: 0,
+})
+
+const tasksRefreshIntervalSeconds = 15
```

Comment:
```suggestion
const TASKS_REFRESH_INTERVAL_SECONDS = 15
```
since for all the other ones we use a similar format (and since it is a hardcoded const), see e.g. `THRESHOLDS_POLLING_INTERVAL`

Reply:
Done